modules/kubernetes-addons/karpenter/locals.tf

```diff
@@ -43,10 +43,13 @@ locals {
     irsa_iam_policies                   = concat([aws_iam_policy.karpenter.arn], var.irsa_policies)
   }
 
-  argocd_gitops_config = {
+  argocd_gitops_config = merge(
+    {
     enable                    = true
     serviceAccountName        = local.service_account
     controllerClusterEndpoint = var.addon_context.aws_eks_cluster_endpoint
     awsDefaultInstanceProfile = var.node_iam_instance_profile
-  }
+  },
+  var.helm_config
```

Comment:
No other addon does this, I mean taking data from `var.helm_config` which currently is only when terraform does does the helm install via the helm_release module.

My impression was that parameters that are not absolutely required from terraform, the user will have the chance to  set in git repo and not go back to terraform when changing something about the addon.


@bryantbiggs @apoo Do we want to this `merge` with `var.helm_config` with all the addosn?

Reply:
This already exists also in the External-dns addon: https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/external-dns/main.tf#L5-L10.

Here it is used to pass additional value created by Terraform (here the SNS queue arn) to GitOps. the helm_config allow this to be dynamic without needed to manually add each field

Reply:
Hi @bryantbiggs I will delegate to you if we want to allow the injection of `var.helm_config` into gitops for one addon.